src/storage/src/source/kafka.rs

```diff
@@ -88,13 +87,12 @@ impl SourceReader for KafkaSourceReader {
         _: AwsExternalId,
         restored_offsets: Vec<(PartitionId, Option<MzOffset>)>,
         _: SourceDataEncoding,
-        base_metrics: SourceBaseMetrics,
+        ctx: SourceContext,
```

Comment:
@guswynn I'm not sure this `SourceContext` type is that valuable? Like, why bundle together the metrics and the librdkafka log level, but none of the other fields?

Reply:
Oh, also, the AWS external ID should totally go in the connector context! I'll fix that tomorrow.

Reply:
yes, I was about to say! after this merges I was gonna move that in!